test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/AddressBinderTests.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions;
+using Microsoft.Extensions.Logging.Abstractions;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Tests
+{
+    public class AddressBinderTests
+    {
+        [Theory]
+        [InlineData("10.10.10.10", "10.10.10.10")]
+        [InlineData("[::1]", "::1")]
+        public void CorrectIPEndpointsAreCreated(string host, string expectedAddress)
+        {
+            Assert.True(AddressBinder.TryCreateIPEndPoint(
+                ServerAddress.FromUrl($"http://{host}:5000/"), out var endpoint));
+            Assert.NotNull(endpoint);
+            Assert.Equal(IPAddress.Parse(expectedAddress), endpoint.Address);
+            Assert.Equal(5000, endpoint.Port);
+        }
+
+        [Theory]
+        [InlineData("*")]
```

Comment:
The test name is confusing then. Let's name it `DefaultsToIPv6AnyOnInvalidIPAddress`, or something like that.